cinema/views.py

```diff
@@ -97,12 +100,25 @@
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
         if self.action == "retrieve":
             return MovieDetailSerializer
-
+        if self.action == "upload_image":
+            return MovieImageSerializer
         return MovieSerializer
 
+    @action(
+        methods=["post"],
+        detail=True,
+        permission_classes=(IsAdminUser,),
+        url_path="upload-image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
The `upload_image` action is implemented to handle image uploads via a separate endpoint. Ensure that the `ImageField` is not accessible via POST requests to `api/cinema/movies/` outside of this endpoint.